### PR TITLE
Fix "too complex" error in belongs expression

### PR DESCRIPTION
**Pull request by elenatorro**
This PR solves this support issue https://github.com/CartoDB/support/issues/2197

Screenshot 2019-11-08 at 17 11 34
Screenshot 2019-11-08 at 17 13 31

### CODE REVIEW

File: src/renderer/viz/expressions/belongs.js [hunk 1]
```diff
@@ -89,6 +96,38 @@ function generateBelongsExpression (name, inlineMaker, jsEval) {
...
+            return {
+                preface: this._prefaceCode(childSources.input.preface + childSources.list.preface + preface),
+                inline: `${funcName}(${childSources.input.inline})`
+            };
+        }
+
+        _getFuncList () {
+            return (elem) => {
+                const x = isNumber(elem) ? `${elem}.0` : elem;
+                return `if (x${this.compare}${x}) { return 1.; }`;
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/belongs.js]: So removing the whole nested IF here with multiple short IFs, right?

File: src/renderer/viz/expressions/belongs.js [hunk 2]
```diff
@@ -100,7 +139,7 @@ function generateBelongsExpression (name, inlineMaker, jsEval) {
                 }
             });
 
-            this.inlineMaker = inlineMaker(this.list.elems);
+            // this.inlineMaker = inlineMaker(this.list.elems);
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/belongs.js]: remove this?

File: src/renderer/viz/expressions/belongs.js [hunk 3]
```diff
@@ -91,8 +91,14 @@ function generateBelongsExpression (name, inlineMaker, jsEval) {
 
         _bindMetadata (meta) {
             super._bindMetadata(meta);
-            checkType(name, 'input', 0, 'category', this.input);
-            checkType(name, 'list', 1, 'category-list', this.list);
+            const validTypes = ['number', 'category', 'date'];
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/belongs.js]: so are we finally supporting also 'dates' here?

File: src/renderer/viz/expressions/belongs.js (hunk 1 above)
Comment (VictorVelarde) [reply src/renderer/viz/expressions/belongs.js]: REALLY COOL! ❤️

File: src/renderer/viz/expressions/belongs.js (hunk 2 above)
Comment (elenatorro) [inline src/renderer/viz/expressions/belongs.js]: Ups! let's clean it up

File: src/renderer/viz/expressions/belongs.js (hunk 3 above)
Comment (elenatorro) [reply src/renderer/viz/expressions/belongs.js]: Sure, good catch, I'm going to remove it for now 👍

File: src/renderer/viz/expressions/belongs.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { implicitCast, checkType, checkExpression, checkMaxArguments } from './utils';
 import BaseExpression from './base';
-import { isNumber } from 'util';
+import * as util from 'util';
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/belongs.js]: just out of curiosity, why importing the whole thing here?

File: src/renderer/viz/expressions/belongs.js (hunk 3 above)
Comment (VictorVelarde) [reply src/renderer/viz/expressions/belongs.js]: I think I was reviewing a previous commit, sorry. 😄

**Comment by VictorVelarde** [comment]
🆒 A super-nice job with this @elenatorro, thx!

**Pull request merged**